source/extensions/filters/http/cache/cacheability_utils.cc

```diff
@@ -69,8 +69,14 @@ bool CacheabilityUtils::isCacheableResponse(const Http::ResponseHeaderMap& heade
       (headers.Date() && response_cache_control.max_age_.has_value()) ||
       headers.get(Http::Headers::get().Expires);
 
+  // TODO(cbdm): only allowing responses to be cacheable if there's no vary header or it is empty;
+  // need to implement the logic for using an allowlist.
```

Comment:
note: maybe vary:accept-encoding can at some point be handled automatically, rather than having to specify it? That can happen before or after other allow-list functionality.

Accept-encoding is 'special' in that probably you'd want to have some policy where you'd decode responses based on the Accept header, however they are stored.

Reply:
That's a good point! I added a TODO comment so we can get to it in the future.
I'm currently working on the allowlist implementation, so I'll hopefully get to this after that.